yunwoo/baekjoon/수학/4153-직각-삼각형.js

```diff
@@ -0,0 +1,37 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  let result = "";
+
+  for (let i = 0; i < input.length; i++) {
+    const [c, a, b] = input[i]
+      .split(" ")
+      .map(Number)
+      .sort((a, b) => b - a);
+
+    if (c === 0) return;
+
+    if (c * c === a * a + b * b) {
+      result = "right";
+    } else {
+      result = "wrong";
+    }
+
+    console.log(result);
+  }
```

Comment:
```suggestion
  let result = "";

  for (let i = 0; i < input.length - 1; i++) {
    const [c, a, b] = input[i]
      .split(" ")
      .map(Number)
      .sort((a, b) => b - a);
    
    const isTriangle = c * c === a * a + b * b;
    
    result += isTriangle ? `right\n` : `wrong\n`;
  }
  
  console.log(result);
```
- 풀이 잘봤습니다. 🙇‍♂️
- for문을 input.length - 1만큼 순회하면 c === 0 if문은 필요없어집니다~!
- if문에 들어가는 조건식이 복잡해보이면 변수로 빼주시는 것을 추천합니다.
- 변수로 빼고 result에다 +=, 삼항 연산자로 간단하게 처리할 수 있습니다.
- 반복해서 말씀드리지만 console.log는 문제 결과 제출 딱 1회에만 호출되게 해주세요 🙏
- sort문에서 사용하는 인자가 a, b라 입력값의 a, b, c가 혼동되는 느낌이 있습니다 삼각형의 변이라고 인지할 수 있는 변수로 바꿔보세요!!